### PR TITLE
Move freestyle to freestyle free

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes https://github.com/frees-io/freestyle/issues/469

### CODE REVIEW

**Comment by raulraja** [comment]
needs to resolve conflicts before we consider the diff for review.

File: core/src/main/scala/implicits.scala [hunk 1]
```diff
@@ -18,9 +18,9 @@ package freestyle.cassandra
 
 import cats.{~>, MonadError}
 import com.datastax.driver.core.{Cluster, Session}
-import freestyle.async.AsyncContext
-import freestyle.async.{Implicits => AsyncImplicits}
-import freestyle.asyncGuava.AsyncGuavaImplicits
+import freestyle.free.async.AsyncContext
```
Comment (raulraja) [inline core/src/main/scala/implicits.scala]: I think the AsyncContext is generic enough that it's like in RPC and Cassandra libs where it does not need a free and a tagless version.

**Review by raulraja — APPROVED** [review]
All good but let's restore the package in `async` to not be nested under `free`

**Review by raulraja — APPROVED** [review]

**Review by fedefernandez — APPROVED** [review]
Thanks

**Pull request merged**